### DISCUSSION TITLE
How to use Currency.js library in Adonis.js?

### DISCUSSION THREAD

**Discussion by Agilulfo1820** [adonisjs/core#2059] [Help] [answered]
Hello everyone, 
I need to use some library to manage currencies (add, sum, floating point, etc.) in Adonis.js v4.

I found different libraries that should work (Diner.js, currency.js, money.js), but I am unable to use those libraries with Adonis.

For example with currency.js, I install it with npm
```
npm install --save currency.js
```
and then inside the controller I tried to use it with:
```
const currency = require('currency.js') //nada, but if I click on it it redirects me to the node modules folder
const currency = use('currency.js') //nada
const currency = require('currency.js/dist') //nada
const currency = require('currency.js/dist/currency.js') //nada
```

It always shows this error:
`
Cannot find module 'currency.js' Require stack: - /srv/app/app/Controllers/Http/Common/AccountTransactionController.js - /srv/app/node_modules/require-stack/src/index.js - /srv/app/node_modules/require-stack/index.js - /srv/app/node_modules/@adonisjs/fold/src/Ioc/index.js - /srv/app/node_modules/@adonisjs/fold/index.js - /srv/app/server.js
`

I tried with the other two libraries and it's the same.
Can you please help me import and use this/those library/libraries?

**Comment by thetutlage** [comment]
> `const currency = require('currency.js') //nada, but if I click on it it redirects me to the node modules folder`

I don't understand what this means.

**Reply by Agilulfo1820** [reply]
Sorry for my bad english. 
What I mean is that the package is correctly installed and Webstorm is able to link it to the correct installed folder inside node_modules.

**Reply by Agilulfo1820** [reply]
I just can't understand how to import this library inside my controller, and use it. I have the same problem with other related libraries (like Dinero.js and money.js)

**Comment by RomainLanz** [comment, marked as answer]
You should require it like any other packages.

```js
const currency = require('currency.js')
```

What error do you have when you are doing this?

**Reply by Agilulfo1820** [reply]
```
Error
Cannot find module 'currency.js' Require stack: - /srv/app/app/Controllers/Http/Common/AccountTransactionController.js - /srv/app/node_modules/require-stack/src/index.js - /srv/app/node_modules/require-stack/index.js - /srv/app/node_modules/@adonisjs/fold/src/Ioc/index.js - /srv/app/node_modules/@adonisjs/fold/index.js - /srv/app/server.js
```

**Reply by thetutlage** [reply]
Have you installed the package?

**Reply by thetutlage** [reply]
Also, `require` is a Node.js method. AdonisJS does nothing to it. So the best way to try and reproduce the issue in a vanilla node app

**Reply by Agilulfo1820** [reply]
You are right. It was my fault from the beginning. I'm using docker-compose as my development setup and even If I installed the library it wasn't saved inside the docker volume.

Excuse me for wasting your time and thank you for helping me.
Have a nice holiday guys!
